dhcp-controller/dhcpserver/pkg/server/server.go

```diff
@@ -0,0 +1,428 @@
+package server
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+
+	//"text/template"
+	"time"
+
+	"crypto/md5"
+	"encoding/csv"
+	"encoding/hex"
+	"encoding/json"
+	"io"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"dhcpserver/pkg/kubernetes"
+
+	"github.com/fsnotify/fsnotify"
+)
+
+var (
+	serverLog   = ctrl.Log.WithName("server")
+	dnsmasqLog  = ctrl.Log.WithName("dnsmasq")
+	leasePath   = "/var/lib/misc/dnsmasq.leases"
+	confFile    = "/etc/dnsmasq.d/dnsmasq.conf"
+	k8sClient   *kubernetes.Client
+	HostNetwork = []net.IP{}
+	Netmask     = []string{}
+	Leasetime   = []string{}
+)
+
+type LeaseFile struct {
+	EpochTimestamp string `json:"epoch-timestamp"`
+	MacAddress     string `json:"mac_addr"`
+	IPAddress      string `json:"ip_addr"`
+	Hostname       string `json:"hostname"`
+	ClientID       string `json:"client-ID"`
+}
+
+func parseConfig() error {
+	f, err := os.Open(confFile)
+	if err != nil {
+		serverLog.Error(err, "Cannot open config file")
+		return err
+	}
+	defer f.Close()
+
+	fileScanner := bufio.NewScanner(f)
+
+	fileScanner.Split(bufio.ScanLines)
+
+	for fileScanner.Scan() {
+		line := fileScanner.Text()
+		if strings.Contains(line, "dhcp-range") {
+			dhcprangearray := regexp.MustCompile("[\\=\\s,]").Split(line, -1)
+
+			// TODO: when adding support for vlan, increment all the indices of dhcprangearray
+			// in this block by 1
```

Comment:
we might need to lookup all the possible config options for dhcp-range in dnsmasq

Reply:
They are given here https://github.com/imp/dnsmasq/blob/master/dnsmasq.conf.example#L162-L220